### PR TITLE
Add Travis CI documentation

### PR DESCRIPTION
**Pull request by cfarm**
Our use of Travis has gotten complex enough that it's not self-evident from the codebase's [.travis.yml](https://github.com/cfpb/cfgov-refresh/blob/master/.travis.yml) or the Travis CI Settings. This page briefly describes what we use Travis for and how we configure it to do those things.

## Additions

- adds new page about How we use Travis CI
- adds note on Development Tips page about editing the docs navigation

## Removals

-

## Changes

- Update note on Development Tips about deploying docs to reflect current setup

## Testing

1. run the docs locally:
    ```sh
    pip install -r requirements/manual.txt
    mkdocs serve
    ```
1. go to http://127.0.0.1:8000/travis/
1. go to http://127.0.0.1:8000/development-tips/

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [ ] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: docs/travis.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# How we use Travis CI
+
+## What Travis does
+We use [Travis CI](https://travis-ci.org/) on cfgov-refresh to perform the following tasks:
+
+- Run automated unit tests, accessibility tests, and acceptance tests.
```
Comment (cfarm) [inline docs/travis.md]: I actually should check if this is true

Comment (chosak) [reply docs/travis.md]: I don't know of any particular accessibility tests. We do run unit and acceptance tests.

File: docs/development-tips.md [hunk 2]
```diff
@@ -103,9 +103,6 @@ may have an impact on local server performance.
 Our documentation is written as Markdown files and served in GitHub pages
 by [mkdocs](https://www.mkdocs.org/user-guide/deploying-your-docs/).
 
-To update the docs in GitHub Pages once a pull request has been merged,
-mkdocs provides [a helpful command](https://www.mkdocs.org/user-guide/deploying-your-docs/):
+Every time a PR is merged to master, Travis will build and deploy the documentation to https://cfpb.github.io/cfgov-refresh.
```
Comment (anselmbradford) [inline docs/development-tips.md]: Should we spell out "pull request (PR)" to avoid jargon in docs?

File: docs/travis.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+- Deploy this documentation website to Github on the [gh-pages branch](https://github.com/cfpb/cfgov-refresh/tree/gh-pages).
```
Comment (anselmbradford) [inline docs/travis.md]: Trivial: `GitHub`

File: docs/travis.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## How Travis is configured
+We use the following constraints to optimize Travis builds for speed and utility:
+
+- Travis runs tests on Pull Requests and subsequent pushes to Pull Request branches only, and not on the master branch.
```
Comment (anselmbradford) [inline docs/travis.md]: Would this be clearer?

> Travis runs tests on Pull Requests only, including subsequent pushed commits after the Pull Request is first opened. It is not run on the `master` branch.

File: docs/travis.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+- Travis deploys documentation on merges and pushes to the master branch only, and not on PRs.
```
Comment (anselmbradford) [inline docs/travis.md]: Trivial: possibly code format `master` so it's clearer its a branch name.

File: docs/travis.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+- We do not run Travis builds of any kind on any other branches that are not master, that are not pull request branches.
```
Comment (anselmbradford) [inline docs/travis.md]: This could probably be merged into point one.

File: docs/travis.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+We use a combination of [build conditionals](https://docs.travis-ci.com/user/conditions-v1) and [build stages](https://docs.travis-ci.com/user/build-stages/) in the [.travis.yml](https://github.com/cfpb/cfgov-refresh/blob/master/.travis.yml) file and Settings in the Travis UI to customize Travis to fit the above constraints.
```
Comment (anselmbradford) [inline docs/travis.md]: Trivial: long line could be broken across lines to make commenting easier.
Should "Settings" be lowercase or in quotes?

**Comment by cfarm** [comment]
@higs4281 I think this is a good place for that except that you list out the satellite repos here, when I would expect that to be listed elsewhere in the docs and we link to it from this page. So more like this:

## An extra task for satellite repositories

For our [seven satellite apps] [link to another page], Travis is also used to build and attach a deployment wheel file to every release.  

An example is the `.whl` file on [this release](https://github.com/cfpb/retirement/releases/tag/0.7.6) of the retirement app.

**Comment by cfarm** [comment]
I'm going to merge the changes I have so far (thanks @anselmbradford for your comments, I made some edits based on your ideas) but may throw together a satellite apps page and mention like Bill suggested in a separate PR

**Pull request merged**